src/extra.ml

```diff
@@ -136,6 +157,37 @@ module IntMap = Map.Make(Int)
 (* Functional maps with [string] keys. *)
 module StrMap = Map.Make(String)
 
+(** Position of a subterm of a term. *)
+module Position =
+struct
+  (** Each element of the list is a level in the tree of the term.  For
+      instance, the subterm [x] in the term [Appl(S, Appl(T, x))] has
+      position [1.2.2], encoded by [[2 ; 2 ; 1]]. *)
+  type t = int list
```

Comment:
I was expecting something like [0;0] instead.
Why not representing the top position by the empty list?
Why not counting from 0 (as in array indexes and List.nth)?
Like this x is the argument of index 0 of Appl(T,x) which itself is the argument of index 0 of Appl(S,Appl(T,x)).